script.js

```diff
@@ -1,4 +1,586 @@
+  }
+  if (gameResult == `lose`) {
+    message = `You chose ${userInput} and I chose ${programSelection}. <br> <br> You lost for this round, but that doesn't mean you'll lose this game. <br> Continue playing to see who the final winner is for this game of Korean SPS!`;
+  }
+  if (gameResult == `win`) {
+    message = `You chose ${userInput} and I chose ${programSelection}. <br> <br> You won for this round, but you're not yet the final winner. <br> Continue playing to see who wins this game of Korean SPS!`;
+  }
+  if (gameResult == `draw` && lastWonPlayer == `not yet assigned`) {
+    message = `You chose ${userInput} and I chose ${programSelection} too! <br> <br> It's a draw for this round, but there weren't any winners from the previous round(s), so there is no final winner yet. <br> <br> Continue playing for a chance to become the final winner!`;
+  }
+  if (gameResult == `draw` && lastWonPlayer != `not yet assigned`) {
+    message = `You chose ${userInput} and I chose ${programSelection} too! <br> <br> It's a draw! Which means... ${lastWonPlayer} the winner!`;
+  }
+
+  return message;
+};
+
+// Specify what happens for the 3 game modes
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = ``;
+
+  if (currentGameMode == `waiting for player name`) {
+    playerName = input;
+    currentGameMode = `standard game`;
+    myOutputValue = `Welcome, ${playerName}! <br> Please type 'scissors', 'paper' or 'stone' to start playing the game with me 😊 <br> <br> You can also type 'korean' to try our korean mode and 'standard' to revert back to the original game. <br> <br> --- What is the korean version of SPS? --- <br> In Korea, a two-player upgraded version exists by the name muk-jji-ppa. After showing their hands, the player with the winning throw shouts "muk-jji-ppa!" upon which both players throw again. If they throw differently (for example, rock and paper, or paper and scissors), whoever wins this second round shouts "muk-jji-ppa!" and thus the play continues until both players throw the same item (for example, rock and rock), at which point whoever was the last winner becomes the actual winner.`;
+  } else if (currentGameMode != `waiting for player name`) {
+    currentGameMode = changeGameMode(input);
+    programChoice = generateChoice();
+    gameOutcome = result(input, programChoice);
+    if (gameOutcome != `draw`) {
+      lastWonPlayer = updateWinner(input, programChoice);
+    }
+    if (currentGameMode == `standard game`) {
+      myOutputValue = standardGame(input, programChoice);
+    }
+    if (currentGameMode == `korean game`) {
+      myOutputValue = koreanGame(
+        input,
+        programChoice,
+        gameOutcome,
+        lastWonPlayer
+      );
```

Comment:
Think these have to be indented to the right

Reply:
Keep up the way you are naming your variables. It is clear, you could explore shortening some, for eg. `currentGameMode` to `currGameMode` - this is minor though

Do explore and utilise scream case variables for upcoming projects.

Overall, I can you do have a good grasp of how the program is flowing from top to bottom, so keep it up Eunice! You're really acing 💪